docs/csharp/tour-of-csharp/features.md

```diff
@@ -0,0 +1,118 @@
+
+:::code language="csharp" source="./snippets/shared/Features.cs" ID="UseDelegate":::
+
+A delegate doesn't know or care about the class of the method it references. All that matters is that the referenced method has the same parameters and return type as the delegate.
+
+## async / await
+
+C# supports asynchronous programs with two keywords: `async` and `await`. You add the `async` modifier to a method declaration to declare the method is asynchronous. The `await` operator tells the compiler that to asynchronously await for a result to finish. Control is returned to the caller, and the method returns a structure that manages the state of the asynchronous work. The structure is typically a <xref:System.Threading.Tasks.Task%601?displayProperty=nameWithType>, but can be any type that supports the awaiter pattern. These features enable you to write code that reads as its synchronous counterpart, but executes asynchronously. For example, the following code downloads the home page for [Microsoft docs](https://docs.microsoft.com):
+
+:::code language="csharp" source="./snippets/shared/Features.cs" ID="AsyncExample":::
+
+This small sample shows the major features for asynchronous programming:
+
+- The method declaration includes the `async` modifier.
+- The body of the method `await`s the return of the `GetByteArrayAsync` method.
+- The type specified in the `return` statement matches the type argument in the `TaskL<T>` declaration for the method. (A method that returns a `Task` would use `return` statements without any argument).
+
+## Attributes
+
+Types, members, and other entities in a C# program support modifiers that control certain aspects of their behavior. For example, the accessibility of a method is controlled using the `public`, `protected`, `internal`, and `private` modifiers. C# generalizes this capability such that user-defined types of declarative information can be attached to program entities and retrieved at run-time. Programs specify this additional declarative information by defining and using [***attributes***](../programming-guide/concepts/attributes/index.md).
+
+The following example declares a `HelpAttribute` attribute that can be placed on program entities to provide links to their associated documentation.
+
+:::code language="csharp" source="./snippets/shared/Features.cs" ID="DefineAttribute":::
+
+All attribute classes derive from the <xref:System.Attribute> base class provided by the standard library. Attributes can be applied by giving their name, along with any arguments, inside square brackets just before the associated declaration. If an attribute’s name ends in `Attribute`, that part of the name can be omitted when the attribute is referenced. For example, the `HelpAttribute` can be used as follows.
```

Comment:
is "the standard library" the right term here?

Reply:
That's an artifact from when it was in the C# standard. Updated.